src/Renci.SshNet/Sftp/SftpFileStream.cs

```diff
@@ -267,6 +292,110 @@ internal SftpFileStream(ISftpSession session, string path, FileMode mode, FileAc
             }
         }
 
+#if FEATURE_TAP
+        internal static async Task<SftpFileStream> OpenAsync(ISftpSession session, string path, FileMode mode, FileAccess access, int bufferSize, CancellationToken cancellationToken)
+        {
+            if (session == null)
+                throw new SshConnectionException("Client not connected.");
+            if (path == null)
+                throw new ArgumentNullException("path");
+            if (bufferSize <= 0)
+                throw new ArgumentOutOfRangeException("bufferSize");
```

Comment:
Add message parameter "bufferSize must not be less than 0" or something like this.